searx/network/client.py

```diff
@@ -78,33 +111,29 @@ def __init__(self, *args, **kwargs):
         # this on purpose if the base class is not called
         pass
 
-    async def handle_async_request(self, request):
+    def handle_request(self, request):
         raise httpx.UnsupportedProtocol('HTTP protocol is disabled')
 
-    async def aclose(self) -> None:
+    def close(self) -> None:
         pass
 
-    async def __aenter__(self):
+    def __enter__(self):  # Use generics for subclass support.
         return self
 
-    async def __aexit__(
-        self,
-        exc_type=None,
-        exc_value=None,
-        traceback=None,
-    ) -> None:
+    def __exit__(self, exc_type, exc_value, traceback) -> None:  # pylint: disable=signature-differs
+        # avoid to import the various type for the signature, but pylint is not happy
         pass
 
 
-class AsyncProxyTransportFixed(AsyncProxyTransport):
-    """Fix httpx_socks.AsyncProxyTransport
+class _ProxyTransportFixed(SyncProxyTransport):
+    """Fix httpx_socks.SyncProxyTransport
```

Comment:
It does not fix SyncProxyTransport, it wraps the python_socks exceptions into httpx exceptions, so other modules deals only with httpx exceptions.

* https://github.com/romis2012/python-socks/blob/master/python_socks/_errors.py
* https://www.python-httpx.org/exceptions/